express-notes-api/index.js

```diff
@@ -0,0 +1,209 @@
+const express = require("express")
+const app = express()
+
+const fs = require("fs")
+
+const data = require("./data.json")
+let {nextId,notes}=data
+
+
+app.get("/api/notes",(req,res)=>{
+  let notesArr = []
+
+  for (const prop in notes){
+    notesArr.push(notes[prop])
+  }
+
+  res.status(200).json(notesArr)
+})
+
+app.get("/api/notes/:id",(req,res)=>{
+  if (isNaN(req.params.id) === false){
+    let id = parseFloat(req.params.id)
+
+    const idArr=[]
+    for (const prop in notes) {
+      idArr.push(parseInt(prop))
+    }
+
+    if (Number.isInteger(id)){
+      if(Math.sign(id)===1){
+        if(idArr.includes(id)){
+          res.status(200).json(notes[id])
+
+        } else {
+          const err2 = {
+            "error": `cannot find note with id ${id}`
+          }
+          res.status(404).json(err2)
+        }
+
+      } else {
+        const err = {
+          "error": "id must be a positive integer"
+        }
+        res.status(400).json(err)
+      }
+
+    } else {
+      const err = {
+        "error": "id must be a positive integer"
+      }
+      res.status(400).json(err)
+    }
+  } else {
+    const err = {
+      "error": "id must be a positive integer"
+    }
+    res.status(400).json(err)
```

Comment:
I would refactor these to reduce the amount of times you are checking the same thing in the data